src/index.jsx

```diff
@@ -1,112 +1,31 @@
 
-/**
- * Temporary check of auth state vs callback access token to help debug and fix any
- * potential problems.
- */
-function checkAuthState(functionName, authenticationState) {
-  const userIdsMatch = !!(
-    callbackAccessToken && callbackAccessToken.user_id &&
-    authenticationState && authenticationState.authentication &&
-    authenticationState.authentication.userId &&
-    callbackAccessToken.user_id === authenticationState.authentication.userId
+App.subscribe(APP_READY, () => {
+  ReactDOM.render(
+    <AppProvider store={configureStore()}>
+      <Header />
+      <ProfileMain />
+      <Footer />
+    </AppProvider>,
+    document.getElementById('root'),
   );
-  const customAttributes = {
-    functionName,
-    callbackAccessToken,
-    authenticationState,
-    userIdsMatch,
-  };
-  NewRelicLoggingService.logInfo(`checkAuthState: ${functionName}: userIdsMatch=${userIdsMatch}`, customAttributes);
-}
-
-/**
- * We need to merge the application configuration with the authentication state
- * so that we can hand it all to the redux store's initializer.
- */
-function createInitialState() {
-  const authenticationState = apiClient.getAuthenticationState();
-  checkAuthState('createInitialState', authenticationState);
-  return Object.assign({}, { configuration }, authenticationState);
-}
-
-function configure() {
-  configureI18n(configuration, messages);
-
-  const { store, history } = configureStore(createInitialState(), configuration.ENVIRONMENT);
-
-  configureLoggingService(NewRelicLoggingService);
-  configureProfileApiService(configuration, apiClient);
-  configureUserAccountApiService(configuration, apiClient);
-  initializeSegment(configuration.SEGMENT_KEY);
-  configureAnalytics({
-    loggingService: NewRelicLoggingService,
-    authApiClient: apiClient,
-    analyticsApiBaseUrl: configuration.LMS_BASE_URL,
-  });
-
-  return {
-    store,
-    history,
-  };
-}
-
-apiClient.ensurePublicOrAuthenticationAndCookies(
-  window.location.pathname,
-  (accessToken) => {
-    callbackAccessToken = accessToken;
-    checkAuthState('ensurePublicOrAuthenticationAndCookies:callback', apiClient.getAuthenticationState());
-
-    const { store, history } = configure();
+});
 
-    ReactDOM.render(<App store={store} history={history} />, document.getElementById('root'));
+App.subscribe(APP_ERROR, () => {
+  ReactDOM.render(<ErrorPage message={App.error.message} />, document.getElementById('root'));
+});
 
-    if (accessToken) {
-      identifyAuthenticatedUser(accessToken.userId);
-    } else {
-      // TODO: Could this ever happen?
-      // - If this did happen, wouldn't we want to raise an error?
-      // - Adding error logging for now.
-      NewRelicLoggingService.logError('Empty accessToken returned from ' +
-        'ensurePublicOrAuthenticationAndCookies callback.');
-      identifyAnonymousUser();
-    }
-    sendPageEvent();
-  },
-);
+initialize({ messages, loggingService: NewRelicLoggingService });
```

Comment:
I like that this is named parameters in an object vs and ordered set.

Reply:
Yeah, that was a late breaking addition. :) Much more readable.